moveit_ros/planning_interface/moveit_cpp/src/moveit_cpp.cpp

```diff
@@ -36,40 +36,35 @@
 
 #include <stdexcept>
 #include <moveit/moveit_cpp/moveit_cpp.h>
-#include <moveit/planning_scene_monitor/current_state_monitor.h>
-#include <moveit/common_planning_interface_objects/common_objects.h>
-
-#include <std_msgs/String.h>
 #include <tf2/utils.h>
 #include <tf2_ros/transform_listener.h>
-#include <ros/console.h>
-#include <ros/ros.h>
+#include <rclcpp/rclcpp.hpp>
 
 namespace moveit
 {
 namespace planning_interface
 {
-constexpr char LOGNAME[] = "moveit_cpp";
+static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit.ros_planning_interface.moveit_cpp");
 constexpr char PLANNING_PLUGIN_PARAM[] = "planning_plugin";
 
-MoveItCpp::MoveItCpp(const ros::NodeHandle& nh, const std::shared_ptr<tf2_ros::Buffer>& tf_buffer)
-  : MoveItCpp(Options(nh), nh, tf_buffer)
+MoveItCpp::MoveItCpp(const rclcpp::Node::SharedPtr& node, const std::shared_ptr<tf2_ros::Buffer>& tf_buffer)
+  : MoveItCpp(node, Options(node), tf_buffer)
 {
 }
 
-MoveItCpp::MoveItCpp(const Options& options, const ros::NodeHandle& /*unused*/,
+MoveItCpp::MoveItCpp(const rclcpp::Node::SharedPtr& node, const Options& options,
                      const std::shared_ptr<tf2_ros::Buffer>& tf_buffer)
-  : tf_buffer_(tf_buffer)
+  : node_(node), tf_buffer_(tf_buffer)
 {
   if (!tf_buffer_)
-    tf_buffer_.reset(new tf2_ros::Buffer());
-  tf_listener_.reset(new tf2_ros::TransformListener(*tf_buffer_));
+    tf_buffer_ = std::make_shared<tf2_ros::Buffer>(node_->get_clock());
```

Comment:
Something to consider: You may need to establish what time of clock is used by the tf_bufffer. Most every example I found in the ros2 code base uses `auto clock = std::make_shared<rclcpp::Clock>(RCL_ROS_TIME))` to ensure that the TF buffer uses the correct clock

Reply:
That's actually a good point. I wasn't aware that system time was the default in the clock constructor. However, nodes always use ROS time, so using `Node::get_clock()` and `Node::now()` is valid everywhere you want to use ROS time.